### PR TITLE
feat(react/hooks): add eslint-plugin-react-hooks to default react conf

### PR DESCRIPTION
**Pull request by Slashgear**
- Add eslint-plugin-react-hooks by default in our react plugin to help use lint our hooks.

I use default configuration of this plugin based on 👍 
https://www.npmjs.com/package/eslint-plugin-react-hooks#installation

### CODE REVIEW

File: packages/eslint-plugin-react/index.js [hunk 1]
```diff
@@ -18,6 +18,9 @@ module.exports = {
         // prefer "<></>" syntax over <React.Fragment></React.Fragment>
         'react/jsx-fragments': ['error', 'syntax'],
 
+        'react-hooks/rules-of-hooks': 'error',
+        'react-hooks/exhaustive-deps': 'warn',
```
Comment (fdubost) [inline packages/eslint-plugin-react/index.js]: Can we use 'error' instead? We dont pay attention at lint warnings 😕

Comment (Slashgear) [reply packages/eslint-plugin-react/index.js]: In some case @flepretre showed me that we don't want to trigger effect on some "dependencies" props. Or in order to make a `didMount` like usage.

Not sure of what to do

Comment (fdubost) [reply packages/eslint-plugin-react/index.js]: Ok, let it as a warning 😕

**Comment by flepretre** [comment]
Thanks for the warning on 'useEffect'.

Le mer. 11 déc. 2019 à 10:24, Florent Dubost <notifications@github.com> a
écrit :

**Pull request merged**